--- lndir-1.0.3.orig/configure.ac	2012-03-08 07:50:53.000000000 +0200
+++ lndir-1.0.3/configure.ac	2014-01-24 16:17:44.622580481 +0200
@@ -33,16 +33,9 @@
 AM_INIT_AUTOMAKE([foreign dist-bzip2])
 AM_MAINTAINER_MODE
 
-# Require X.Org macros 1.8 or later for MAN_SUBSTS set by XORG_MANPAGE_SECTIONS
-m4_ifndef([XORG_MACROS_VERSION],
-	  [m4_fatal([must install xorg-macros 1.8 or later before running autoconf/autogen])])
-XORG_MACROS_VERSION(1.8)
-XORG_DEFAULT_OPTIONS
 
 # Avoid stat() failures if a file has a size that doesn't fit in 32-bits
 AC_SYS_LARGEFILE
 
-# Checks for pkg-config packages
-PKG_CHECK_MODULES(XPROTO, xproto >= 7.0.17)
 
 AC_OUTPUT([Makefile])
 
--- lndir-1.0.3.orig/Makefile.am	2012-03-08 07:50:53.000000000 +0200
+++ lndir-1.0.3/Makefile.am	2014-01-24 16:11:06.870574046 +0200
@@ -20,7 +20,6 @@
 #  PERFORMANCE OF THIS SOFTWARE.
 
 bin_PROGRAMS = lndir
-AM_CFLAGS = $(CWARNFLAGS) $(XPROTO_CFLAGS)
 lndir_SOURCES = lndir.c
 
 # Man page
@@ -28,7 +27,7 @@
 
 appman_PRE = lndir.man
 
-appman_DATA = $(appman_PRE:man=@APP_MAN_SUFFIX@)
+appman_DATA = $(appman_PRE)
 
 CLEANFILES = $(appman_DATA)
 
--- lndir-1.0.3.orig/lndir.c	2012-03-08 07:50:53.000000000 +0200
+++ lndir-1.0.3/lndir.c	2014-01-24 16:43:42.434605688 +0200
@@ -49,8 +49,7 @@
 #include "config.h"
 #endif
 
-#include <X11/Xos.h>
-#include <X11/Xfuncproto.h>
+#include <string.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <sys/stat.h>
@@ -72,7 +71,7 @@
 char *rcurdir;
 char *curdir;
 
-static void _X_ATTRIBUTE_PRINTF(2,3) _X_NORETURN
+static void __attribute__((__format__(__printf__,2,3))) __attribute((noreturn))
 quit (int code, const char * fmt, ...)
 {
     va_list args;
@@ -83,14 +82,14 @@
     exit (code);
 }
 
-static void _X_NORETURN
+static void __attribute((noreturn))
 quiterr (int code, const char *s)
 {
     perror (s);
     exit (code);
 }
 
-static void _X_ATTRIBUTE_PRINTF(1,2)
+static void  __attribute__((__format__(__printf__,1,2)))
 msg (const char * fmt, ...)
 {
     va_list args;
