--- stm-target-openssl.spec.orig	2014-04-27 17:17:16.830075265 +0200
+++ stm-target-openssl.spec	2014-04-27 17:18:24.206075537 +0200
@@ -4,6 +4,7 @@
 Release: 30
 Source0: ftp://ftp.openssl.org/source/openssl-%{version}.tar.gz
 Patch0: openssl-0.9.8g-hwrandom.patch
+Patch1: stm-target-openssl-1.0.1g-fix_pod_syntax.patch
 
 License: Freely distributable
 Group: System Environment/Libraries
@@ -98,6 +99,7 @@
 %prep
 %setup -q -n openssl-%{version}
 %patch0 -p1
+%patch1 -p1
 
 %build
 %target_setup
