--- stm-target-usbutils.spec.org	2013-12-19 00:27:39.962145121 +0200
+++ stm-target-usbutils.spec	2013-12-19 00:27:56.966229442 +0200
@@ -6,7 +6,6 @@
 URL		: http://packages.debian.org/stable/utils/usbutils
 License 	: GPL 
 Group	  	: Applications/System
-BuildRequires   : %{_stm_pkg_prefix}-%{_stm_target_arch}-libusb-dev
 Buildroot 	: %(mktemp -ud %{_tmppath}/%{name}-%{version}-%{release}-%{_stm_target_arch}-XXXXXX)
 Prefix	  	: %{_stm_cross_target_dir}
 
