diff -ruN binutils-2.20.1-baseline/bfd/po/Make-in binutils-2.20.1/bfd/po/Make-in
--- binutils-2.20.1-baseline/bfd/po/Make-in	2009-09-02 00:18:38.000000000 -0700
+++ binutils-2.20.1/bfd/po/Make-in	2010-03-15 01:05:08.404273334 -0700
@@ -20,10 +20,10 @@
 
 prefix = @prefix@
 exec_prefix = @exec_prefix@
-datadir = $(prefix)/@DATADIRNAME@
+datadir = @datadir@
 localedir = $(datadir)/locale
-gnulocaledir = $(prefix)/share/locale
-gettextsrcdir = $(prefix)/share/gettext/po
+gnulocaledir = $(datadir)/locale
+gettextsrcdir = $(datadir)/gettext/po
 subdir = po
 
 DESTDIR =
diff -ruN binutils-2.20.1-baseline/binutils/po/Make-in binutils-2.20.1/binutils/po/Make-in
--- binutils-2.20.1-baseline/binutils/po/Make-in	2009-09-02 00:22:32.000000000 -0700
+++ binutils-2.20.1/binutils/po/Make-in	2010-03-15 01:05:08.404273334 -0700
@@ -20,10 +20,10 @@
 
 prefix = @prefix@
 exec_prefix = @exec_prefix@
-datadir = $(prefix)/@DATADIRNAME@
+datadir = @datadir@
 localedir = $(datadir)/locale
-gnulocaledir = $(prefix)/share/locale
-gettextsrcdir = $(prefix)/share/gettext/po
+gnulocaledir = $(datadir)/locale
+gettextsrcdir = $(datadir)/gettext/po
 subdir = po
 
 DESTDIR =

diff -ruN binutils-2.20.1-baseline/gas/po/Make-in binutils-2.20.1/gas/po/Make-in
--- binutils-2.20.1-baseline/gas/po/Make-in	2009-09-02 00:24:21.000000000 -0700
+++ binutils-2.20.1/gas/po/Make-in	2010-03-15 01:05:08.405880796 -0700
@@ -20,10 +20,10 @@
 
 prefix = @prefix@
 exec_prefix = @exec_prefix@
-datadir = $(prefix)/@DATADIRNAME@
+datadir = @datadir@
 localedir = $(datadir)/locale
-gnulocaledir = $(prefix)/share/locale
-gettextsrcdir = $(prefix)/share/gettext/po
+gnulocaledir = $(datadir)/locale
+gettextsrcdir = $(datadir)/gettext/po
 subdir = po
 
 DESTDIR =
diff -ruN binutils-2.20.1-baseline/gprof/po/Make-in binutils-2.20.1/gprof/po/Make-in
--- binutils-2.20.1-baseline/gprof/po/Make-in	2009-09-02 00:24:26.000000000 -0700
+++ binutils-2.20.1/gprof/po/Make-in	2010-03-15 01:05:08.405880796 -0700
@@ -20,10 +20,10 @@
 
 prefix = @prefix@
 exec_prefix = @exec_prefix@
-datadir = $(prefix)/@DATADIRNAME@
+datadir = @datadir@
 localedir = $(datadir)/locale
-gnulocaledir = $(prefix)/share/locale
-gettextsrcdir = $(prefix)/share/gettext/po
+gnulocaledir = $(datadir)/locale
+gettextsrcdir = $(datadir)/gettext/po
 subdir = po
 
 DESTDIR =

diff -ruN binutils-2.20.1-baseline/ld/po/Make-in binutils-2.20.1/ld/po/Make-in
--- binutils-2.20.1-baseline/ld/po/Make-in	2009-09-02 00:25:35.000000000 -0700
+++ binutils-2.20.1/ld/po/Make-in	2010-03-15 01:05:08.409873651 -0700
@@ -20,10 +20,10 @@
 
 prefix = @prefix@
 exec_prefix = @exec_prefix@
-datadir = $(prefix)/@DATADIRNAME@
+datadir = @datadir@
 localedir = $(datadir)/locale
-gnulocaledir = $(prefix)/share/locale
-gettextsrcdir = $(prefix)/share/gettext/po
+gnulocaledir = $(datadir)/locale
+gettextsrcdir = $(datadir)/gettext/po
 subdir = po
 
 DESTDIR =
diff -ruN binutils-2.20.1-baseline/opcodes/po/Make-in binutils-2.20.1/opcodes/po/Make-in
--- binutils-2.20.1-baseline/opcodes/po/Make-in	2009-09-02 00:20:30.000000000 -0700
+++ binutils-2.20.1/opcodes/po/Make-in	2010-03-15 01:05:08.409873651 -0700
@@ -20,10 +20,10 @@
 
 prefix = @prefix@
 exec_prefix = @exec_prefix@
-datadir = $(prefix)/@DATADIRNAME@
+datadir = @datadir@
 localedir = $(datadir)/locale
-gnulocaledir = $(prefix)/share/locale
-gettextsrcdir = $(prefix)/share/gettext/po
+gnulocaledir = $(datadir)/locale
+gettextsrcdir = $(datadir)/gettext/po
 subdir = po
 
 DESTDIR =

diff -ruN binutils-2.20.1-baseline/gold/po/Make-in binutils-2.20.1/gold/po/Make-in
--- binutils-2.20.1-baseline/gold/po/Make-in	2009-09-02 00:20:30.000000000 -0700
+++ binutils-2.20.1/gold/po/Make-in	2010-03-15 01:05:08.409873651 -0700
@@ -20,10 +20,10 @@

 prefix = @prefix@
 exec_prefix = @exec_prefix@
-datadir = $(prefix)/@DATADIRNAME@
+datadir = @datadir@
 localedir = $(datadir)/locale
-gnulocaledir = $(prefix)/share/locale
-gettextsrcdir = $(prefix)/share/gettext/po
+gnulocaledir = $(datadir)/locale
+gettextsrcdir = $(datadir)/gettext/po
 subdir = po

 DESTDIR =
