diff --git a/input/input.c b/input/input.c
index 349e6d0..93406ba 100644
--- a/input/input.c
+++ b/input/input.c
@@ -8,14 +8,14 @@
 #include "gfx.h"
 #include "inputd.h"
 
-#define NCF_FILE 	"/var/tuxbox/config/neutrino.conf"
+#define NCF_FILE CONFIGDIR	"/neutrino.conf"
 #define BUFSIZE 	1024
 #define I_VERSION	1.08
 
 //#define FONT "/usr/share/fonts/md_khmurabi_10.ttf"
-#define FONT2 "/share/fonts/pakenham.ttf"
+#define FONT2 FONTDIR "/pakenham.ttf"
 // if font is not in usual place, we look here:
-#define FONT "/share/fonts/neutrino.ttf"
+#define FONT FONTDIR "/neutrino.ttf"
 
 //					   CMCST,   CMCS,  CMCT,    CMC,    CMCIT,  CMCI,   CMHT,   CMH
 //					   WHITE,   BLUE0, TRANSP,  CMS,    ORANGE, GREEN,  YELLOW, RED
diff --git a/msgbox/msgbox.c b/msgbox/msgbox.c
index 14aa0e7..01eb32e 100644
--- a/msgbox/msgbox.c
+++ b/msgbox/msgbox.c
@@ -10,15 +10,15 @@
 
 #define M_VERSION 1.12
 
-#define NCF_FILE 	"/var/tuxbox/config/neutrino.conf"
+#define NCF_FILE CONFIGDIR  "/neutrino.conf"
 #ifndef MARTII
 #define HDF_FILE	"/tmp/.msgbox_hidden"
 #endif
 
 //#define FONT "/usr/share/fonts/md_khmurabi_10.ttf"
-#define FONT2 "/share/fonts/pakenham.ttf"
+#define FONT2 FONTDIR "/pakenham.ttf"
 // if font is not in usual place, we look here:
-#define FONT "/share/fonts/neutrino.ttf"
+#define FONT FONTDIR "/neutrino.ttf"
 
 //					   CMCST,   CMCS,  CMCT,    CMC,    CMCIT,  CMCI,   CMHT,   CMH
 //					   WHITE,   BLUE0, TRANSP,  CMS,    ORANGE, GREEN,  YELLOW, RED
diff --git a/shellexec/shellexec.c b/shellexec/shellexec.c
index 6f2762a..8acb260 100644
--- a/shellexec/shellexec.c
+++ b/shellexec/shellexec.c
@@ -6,15 +6,15 @@
 #include "io.h"
 #include "gfx.h"
 
-static char CFG_FILE[128]="/var/tuxbox/config/shellexec.conf";
+static char CFG_FILE[128] ="/var/tuxbox/config/shellexec.conf";
 
 //#define FONT "/usr/share/fonts/md_khmurabi_10.ttf"
-#define FONT2 "/share/fonts/pakenham.ttf"
+#define FONT2 FONTDIR "/pakenham.ttf"
 // if font is not in usual place, we look here:
 #ifdef MARTII
-char FONT[128]="/share/fonts/neutrino.ttf";
+char FONT[128] ="/usr/share/fonts/neutrino.ttf";
 #else
-unsigned char FONT[128]="/share/fonts/neutrino.ttf";
+unsigned char FONT[128] ="/usr/share/fonts/neutrino.ttf";
 #endif
 
 //					  CMCST,    CMCS,   CMCT,   CMC,    CMCIT,  CMCI,   CMHT,   CMH
diff --git a/shellexec/shellexec.conf b/shellexec/shellexec.conf
index 7c8016d..b5f9939 100644
--- a/shellexec/shellexec.conf
+++ b/shellexec/shellexec.conf
@@ -1,6 +1,6 @@
 # config
-#FONT=/share/fonts/micron_bold.ttf
-FONT=/share/fonts/neutrino.ttf
+#FONT=/usr/share/fonts/micron_bold.ttf
+FONT=/usr/share/fonts/neutrino.ttf
 FONTSIZE=28
 LINESPP=20
 PAGING=1
diff --git a/tuxcom/tuxcom.h b/tuxcom/tuxcom.h
index 9ce86a1..0979345 100644
--- a/tuxcom/tuxcom.h
+++ b/tuxcom/tuxcom.h
@@ -289,9 +289,9 @@ int rcaltgrtable[] =
 
 //freetype stuff
 
-#define FONT "/share/fonts/neutrino.ttf"
+#define FONT FONTDIR "/neutrino.ttf"
 // if font is not in usual place, we look here:
-#define FONT2 "/share/fonts/pakenham.ttf"
+#define FONT2 FONTDIR "/pakenham.ttf"
 
 enum {LANG_INT,LANG_DE, LANG_IT, LANG_SV, LANG_PT};
 enum {RC_NORMAL,RC_EDIT};
diff --git a/tuxwetter/bmps.c b/tuxwetter/bmps.c
index d8f4db4..d84ecee 100644
--- a/tuxwetter/bmps.c
+++ b/tuxwetter/bmps.c
@@ -64,7 +64,7 @@ int bmp2lcd (char *bildfile)
 		strcpy(filename,bildfile);
 	}
 
-	sprintf(bmpfile,"/share/tuxbox/tuxwetter/%s",filename);
+	sprintf(bmpfile,"/usr/share/tuxbox/tuxwetter/%s",filename);
 	
 	FILE *fbmp;
 
diff --git a/tuxwetter/tuxwetter.c b/tuxwetter/tuxwetter.c
index 0895453..96382ae 100644
--- a/tuxwetter/tuxwetter.c
+++ b/tuxwetter/tuxwetter.c
@@ -131,9 +131,9 @@ static char menucoltxt[][25]={"Content_Selected_Text","Content_Selected","Conten
 //static char spres[][5]={"","_crt","_lcd"};
 
 //#define FONT "/usr/share/fonts/md_khmurabi_10.ttf"
-#define FONT2 "/share/fonts/pakenham.ttf"
+#define FONT2 FONTDIR "/pakenham.ttf"
 // if font is not in usual place, we look here:
-#define FONT "/share/fonts/neutrino.ttf"
+#define FONT FONTDIR "/neutrino.ttf"
 
 //					    CMCST,  CMCS,   CMCT,   CMC,    CMCIT,  CMCI,   CMHT,   CMH
 //					    WHITE,  BLUE0,  TRANSP, CMS,    ORANGE, GREEN,  YELLOW, RED
