--- a/oauth2client/tools.py	2015-05-09 18:57:55.198822254 +0300
+++ b/oauth2client/tools.py	2015-05-10 00:40:28.308739612 +0300
@@ -52,7 +52,11 @@
     import argparse
   except ImportError:
     return None
-  parser = argparse.ArgumentParser(add_help=False)
+  try:
+    prog = sys.argv[0]
+  except:
+    prog = "youtube"
+  parser = argparse.ArgumentParser(prog, add_help=False)
   parser.add_argument('--auth_host_name', default='localhost',
                       help='Hostname when running a local web server.')
   parser.add_argument('--noauth_local_webserver', action='store_true',
