--- stm-target-glibc.spec.orig	2014-12-25 01:03:16.574541781 +0100
+++ stm-target-glibc.spec	2014-12-25 01:06:16.058535997 +0100
@@ -19,6 +19,7 @@
 Source1: glibc-ports-%{version}.tar.bz2
 Source2: glibc-%{version}.changelog
 Source3: glibc-ports-%{version}.changelog
+Patch0: make-versions-4.0-and-greater.patch
 
 %description
 The GNU C Library including standard libraries used by nearly all
@@ -130,6 +131,7 @@
 ##############################################################################
 %prep
 %setup -n glibc-%{version} -q
+%patch0 -p1
 
 tar xjf %SOURCE1
 mv glibc-ports-%{version} ports
