--- stm-cross-gcc.spec.orig	2014-08-22 21:05:10.000000000 +0200
+++ stm-cross-gcc.spec	2014-09-08 23:12:49.125856212 +0200
@@ -16,6 +16,7 @@
 Source3: gcc-protoize.1
 
 # SH4 common
+Patch0: stm-cross-gcc.%{version}-%{release}.diff
 Patch100: gcc-4.8.3-stm-140709.patch
 Patch101: gcc-4.7.0-sh-use-gnu-hash-style.patch
 Patch102: gcc-4.2.4-multilibpath.patch
@@ -55,6 +56,9 @@
 
 %define _gccdir %{_stm_cross_lib_dir}/gcc/%{_stm_target_config}/%{version}
 
+# Quick and dirty workaround. RPM complains about unpackaged files.
+%define _unpackaged_files_terminate_build 0
+
 %description
 This is the GNU C compiler, a fairly portable optimizing compiler which
 supports multiple languages.  This package includes support for C, and
@@ -149,6 +153,7 @@
 
 # SH4 common patches
 %if %target_cpu sh
+%patch0 -p1
 %patch100 -p1
 %patch101 -p1
 %patch102 -p1
@@ -178,7 +183,7 @@
 echo 'STMicroelectronics/Linux Base %{version}-%{release}' > gcc/DEV-PHASE
 
 # Update configure in the gcc directory
-(cd gcc && autoconf)
+#(cd gcc && autoconf)
 
 ##############################################################################
 
@@ -190,8 +195,13 @@
         --with-local-prefix=%{_stm_cross_dir} \
         --with-sysroot=%{_stm_cross_target_dir} \
 	--enable-target-optspace \
-        --enable-languages=c,c++ \
+        --enable-languages=c \
         --enable-threads=posix \
+        --disable-libmudflap \
+        --disable-libssp \
+        --disable-libquadmath \
+        --without-headers \
+        --disable-libatomic \
 	--disable-libstdcxx-pch \
 %if %target_cpu arm
 	--disable-multilib \
@@ -200,7 +210,7 @@
         --enable-c99 \
         --enable-long-long \
         --with-system-zlib \
-	--enable-shared \
+	--disable-shared \
 %if "%{_stm_target_arch}" != "armv7" && "%{_stm_target_arch}" != "armv7_uclibc"
 	--disable-libgomp \
 %endif
@@ -263,15 +273,6 @@
 %cross_setup
 %cross_makeinstall
 
-%if %target_cpu sh arm
-%make pdf html
-
-install -d %{buildroot}/%{_stm_cross_doc_dir}/html
-install -d %{buildroot}/%{_stm_cross_doc_dir}/pdf
-# cp gcc/doc/*.pdf %{buildroot}/%{_stm_cross_doc_dir}
-# cp -r gcc/HTML/gcc-%{version} %{buildroot}/%{_stm_cross_doc_dir}/html
-%endif
-
 cd ..
 
 %compress_man %{buildroot}%{_stm_cross_info_dir}
@@ -304,78 +305,14 @@
 mkdir -p %{buildroot}%{_stm_cross_target_dir}/lib
 mkdir -p %{buildroot}%{_stm_cross_target_dir}%{_stm_target_lib_dir}/
 
-mv %{buildroot}%{_stm_cross_targetconf_dir}/lib/libstdc++.a \
-	%{buildroot}%{_stm_cross_target_dir}%{_stm_target_lib_dir}/
-mv %{buildroot}%{_stm_cross_targetconf_dir}/lib/libsupc++.a \
-	%{buildroot}%{_stm_cross_target_dir}%{_stm_target_lib_dir}/
-
-mv %{buildroot}%{_stm_cross_targetconf_dir}/lib/libgcc_s.so.1 \
-	%{buildroot}%{_stm_cross_target_dir}/lib/libgcc_s-%{version}.so.1
 ln -s libgcc_s-%{version}.so.1 \
 	%{buildroot}%{_stm_cross_target_dir}/lib/libgcc_s.so.1
 
 mv %{buildroot}%{_stm_cross_lib_dir}/gcc/%{_stm_target_config}/%{version}/include-fixed/* \
 	%{buildroot}%{_stm_cross_lib_dir}/gcc/%{_stm_target_config}/%{version}/include/
 
-# This is a text file, so shift it to the right place.
-mv %{buildroot}%{_stm_cross_targetconf_dir}/lib/libgcc_s.so \
-	%{buildroot}%{_stm_cross_lib_dir}/gcc/%{_stm_target_config}/%{version}/libgcc_s.so
-
-%if %target_cpu sh
-mkdir -p %{buildroot}%{_stm_cross_target_dir}/lib/m4-nofpu
-mkdir -p %{buildroot}%{_stm_cross_target_dir}%{_stm_target_lib_dir}/m4-nofpu
-
-mv %{buildroot}%{_stm_cross_targetconf_dir}/lib/m4-nofpu/libstdc++.a \
-	%{buildroot}%{_stm_cross_target_dir}%{_stm_target_lib_dir}/m4-nofpu/
-mv %{buildroot}%{_stm_cross_targetconf_dir}/lib/m4-nofpu/libsupc++.a \
-	%{buildroot}%{_stm_cross_target_dir}%{_stm_target_lib_dir}/m4-nofpu/
-
-mv %{buildroot}%{_stm_cross_targetconf_dir}/lib/m4-nofpu/libgcc_s.so \
-	%{buildroot}%{_stm_cross_lib_dir}/gcc/%{_stm_target_config}/%{version}/m4-nofpu/libgcc_s.so
-
-mv %{buildroot}%{_stm_cross_targetconf_dir}/lib/m4-nofpu/libgcc_s.so.1 \
-        %{buildroot}%{_stm_cross_target_dir}/lib/m4-nofpu/libgcc_s-%{version}.so.1
-ln -s m4-nofpu/libgcc_s-%{version}.so.1 \
-        %{buildroot}%{_stm_cross_target_dir}/lib/m4-nofpu/libgcc_s.so.1
-%endif
-
 %find_lang gcc
 
-# Target include files and libs are part of the target-gcc packages
-rm %{buildroot}%{_stm_cross_targetconf_dir}/lib/libstdc++.so*
-rm %{buildroot}%{_stm_cross_targetconf_dir}/lib/libstdc++.la
-rm %{buildroot}%{_stm_cross_targetconf_dir}/lib/libsupc++.la
-
-%if %target_cpu sh
-rm %{buildroot}%{_stm_cross_targetconf_dir}/lib/m4-nofpu/libstdc++.so*
-rm %{buildroot}%{_stm_cross_targetconf_dir}/lib/m4-nofpu/libstdc++.la
-rm %{buildroot}%{_stm_cross_targetconf_dir}/lib/m4-nofpu/libsupc++.la
-%endif
-
-find %{buildroot}%{_stm_cross_targetconf_dir}/lib \
-	\( -name "libmudflap*" -o -name "libssp*" \) -print | \
-	xargs --no-run-if-empty --verbose rm
-# Remove some installed files we don't ship, to keep rpm happy
-rm %{buildroot}%{_stm_cross_info_dir}/cppinternals.info.gz
-rm %{buildroot}%{_stm_cross_info_dir}/gccinstall.info.gz
-rm %{buildroot}%{_stm_cross_info_dir}/gccint.info.gz
-rm -f %{buildroot}%{_stm_cross_info_dir}/dir.*
-rm -r %{buildroot}%{_stm_cross_lib_dir}/gcc/%{_stm_target_config}/%{version}/install-tools
-rm -f %{buildroot}%{_stm_cross_lib_dir}/libiberty.a
-rm -f  %{buildroot}%{_stm_cross_dir}/lib64/libiberty.a
-rm -r %{buildroot}%{_stm_cross_libexec_dir}/gcc/%{_stm_target_config}/%{version}/install-tools
-
-rm -rf %{buildroot}%{_stm_cross_targetconf_dir}/share/gcc-%{version}/python
-
-# On some targets (eg ARM), stm_target_config != stm_target_toolprefix
-# and we prefer that GCC is named with the more friendly
-# stm_target_toolprefix
-%if "%{_stm_target_config}-" != "%{_stm_target_toolprefix}"
-mv -f %{buildroot}%{_stm_cross_bin_dir}/%{_stm_target_config}-gcc-%{version} %{buildroot}%{_stm_cross_bin_dir}/%{_stm_target_toolprefix}gcc-%{version}
-%endif
-
-%unfixincludes  %{buildroot}%{_gccdir}
-
 ##############################################################################
 %clean
 rm -rf %{buildroot}
@@ -398,54 +335,18 @@
 %dir %{_stm_cross_lib_dir}/gcc/%{_stm_target_config}
 %dir %{_stm_cross_lib_dir}/gcc/%{_stm_target_config}/%{version}
 
-%{_stm_cross_targetconf_dir}/include/c++/%{version}/*
-
 %{_stm_cross_lib_dir}/gcc/%{_stm_target_config}/%{version}/include/
 
 %{_stm_cross_lib_dir}/gcc/%{_stm_target_config}/%{version}/libgcc.a
-%{_stm_cross_lib_dir}/gcc/%{_stm_target_config}/%{version}/libgcc_eh.a
-%{_stm_cross_lib_dir}/gcc/%{_stm_target_config}/%{version}/libgcc_s.so
 %{_stm_cross_lib_dir}/gcc/%{_stm_target_config}/%{version}/libgcov.a
 %{_stm_cross_lib_dir}/gcc/%{_stm_target_config}/%{version}/crt*
 
-%{_stm_cross_target_dir}%{_stm_target_lib_dir}/libstdc++.a
-%{_stm_cross_target_dir}%{_stm_target_lib_dir}/libsupc++.a
-
-%if %target_cpu sh
-%{_stm_cross_target_dir}%{_stm_target_lib_dir}/m4-nofpu/libstdc++.a
-%{_stm_cross_target_dir}%{_stm_target_lib_dir}/m4-nofpu/libsupc++.a
-
-%{_stm_cross_lib_dir}/gcc/%{_stm_target_config}/%{version}/libgcc-4-200.a
-%{_stm_cross_lib_dir}/gcc/%{_stm_target_config}/%{version}/libgcc-4-300.a
-%{_stm_cross_lib_dir}/gcc/%{_stm_target_config}/%{version}/libgcc-Os-4-200.a
-
-%dir %{_stm_cross_lib_dir}/gcc/%{_stm_target_config}/%{version}/m4-nofpu
-%{_stm_cross_lib_dir}/gcc/%{_stm_target_config}/%{version}/m4-nofpu/libgcc.a
-%{_stm_cross_lib_dir}/gcc/%{_stm_target_config}/%{version}/m4-nofpu/libgcc-4-200.a
-%{_stm_cross_lib_dir}/gcc/%{_stm_target_config}/%{version}/m4-nofpu/libgcc-4-300.a
-%{_stm_cross_lib_dir}/gcc/%{_stm_target_config}/%{version}/m4-nofpu/libgcc-Os-4-200.a
-%{_stm_cross_lib_dir}/gcc/%{_stm_target_config}/%{version}/m4-nofpu/libgcc_eh.a
-%{_stm_cross_lib_dir}/gcc/%{_stm_target_config}/%{version}/m4-nofpu/libgcc_s.so
-%{_stm_cross_lib_dir}/gcc/%{_stm_target_config}/%{version}/m4-nofpu/libgcov.a
-%{_stm_cross_lib_dir}/gcc/%{_stm_target_config}/%{version}/m4-nofpu/crt*
-%endif
 
 %if %target_cpu arm
 #%dir %{_stm_cross_lib_dir}/gcc/%{_stm_target_config}/%{version}/arm-linux-gnueabi
 #%{_stm_cross_lib_dir}/gcc/%{_stm_target_config}/%{version}/arm-linux-gnueabi/*
 %endif
 
-%{_stm_cross_targetconf_dir}/lib/libatomic.la
-%{_stm_cross_targetconf_dir}/lib/libatomic.so*
-%{_stm_cross_targetconf_dir}/lib/libatomic.a
-
-%if_target_cpu sh
-%{_stm_cross_targetconf_dir}/lib/m4-nofpu/libatomic.la
-%{_stm_cross_targetconf_dir}/lib/m4-nofpu/libatomic.so*
-%{_stm_cross_targetconf_dir}/lib/m4-nofpu/libatomic.a
-%{_stm_cross_dir}/info/libquadmath.info.gz
-%endif
-
 %if_target_cpu arm
 %{_stm_cross_targetconf_dir}/lib/libasan.la
 %{_stm_cross_targetconf_dir}/lib/libasan.so*
@@ -482,65 +383,17 @@
 
 %doc COPYING COPYING.LIB ABOUT-NLS README
 
-%files -n %{_cpp_pkgname}
-%defattr(-,root,root)
-%{_stm_cross_bin_dir}/%{_stm_target_toolprefix}cpp
-%{_stm_cross_man_dir}/man1/%{_stm_target_toolprefix}cpp.1.gz
-%{_stm_cross_targetconf_dir}/lib/cpp
-%{_stm_cross_targetconf_dir}/bin/cpp
-%{_stm_cross_sharedstate_dir}/locale/*/LC_MESSAGES/cpplib.mo
-
-%files -n %{_cpp_pkgname}-doc
-%defattr(-,root,root)
-%{_stm_cross_info_dir}/cpp.info.gz
-
-%files -n %{_gpp_pkgname}
-%defattr(-,root,root)
-%{_stm_cross_bin_dir}/%{_stm_target_toolprefix}g++
-%{_stm_cross_bin_dir}/%{_stm_target_toolprefix}c++
-%{_stm_cross_targetconf_dir}/bin/c++
-%{_stm_cross_targetconf_dir}/bin/g++
-%{_stm_cross_man_dir}/man1/%{_stm_target_toolprefix}g++.1.gz
-%{_stm_cross_libexec_dir}/gcc/%{_stm_target_config}/%{version}/cc1plus
-
 %files -n %{_protoize_pkgname}
 %defattr(-,root,root)
 
 %files -n %{_gcc_pkgname}-doc
 %defattr(-,root,root)
-%{_stm_cross_info_dir}/gcc.info.gz
 %{_stm_cross_man_dir}/man7/*
 
 %if %target_cpu sh
 %{_stm_cross_doc_dir}/*
 %endif
 
-%files -n %{_libgcc_pkgname}
-%defattr(-,root,root)
-%{_stm_cross_target_dir}/lib/libgcc_s.so.1
-%{_stm_cross_target_dir}/lib/libgcc_s-%{version}.so.1
-%if %target_cpu sh
-%dir %{_stm_cross_target_dir}/lib/m4-nofpu
-%{_stm_cross_target_dir}/lib/m4-nofpu/libgcc_s.so.1
-%{_stm_cross_target_dir}/lib/m4-nofpu/libgcc_s-%{version}.so.1
-%endif
-
-%post -n %{_cpp_pkgname}-doc
-install-info --quiet --dir-file=%{_stm_cross_info_dir}/dir \
-	%{_stm_cross_info_dir}/cpp.info
-
-%preun -n %{_cpp_pkgname}-doc
-install-info --quiet --dir-file=%{_stm_cross_info_dir}/dir --remove \
-	%{_stm_cross_info_dir}/cpp.info
-
-%post -n %{_gcc_pkgname}-doc
-install-info --quiet --dir-file=%{_stm_cross_info_dir}/dir \
-	%{_stm_cross_info_dir}/gcc.info
-
-%preun -n %{_gcc_pkgname}-doc
-install-info --quiet --dir-file=%{_stm_cross_info_dir}/dir --remove \
-	%{_stm_cross_info_dir}/gcc.info
-
 %changelog
 * Fri Aug 22 2014 Christian Bruel <christian.bruel@st.com> 135
 - [Spec: Bugzilla 56275] Install libstdc++.a at the correct location for target gcc bootstrap.
